src/uucore/src/lib/features/selinux.rs

```diff
@@ -187,19 +208,16 @@ mod tests {
             assert!(true, "Successfully set SELinux context");
         } else {
             let err = result.unwrap_err();
-            let valid_errors = [
-                "SELinux is not enabled on this system",
-                &format!(
-                    "Failed to set default context: selinux_lsetfilecon_default() failed on path '{}'",
-                    path.display()
-                ),
-            ];
-
-            assert!(
-                valid_errors.contains(&err.as_str()),
-                "Unexpected error message: {}",
-                err
-            );
+            match err {
+                SeLinuxError::SELinuxNotEnabled => {
+                    assert!(true, "SELinux is not enabled, which is a valid error case");
+                }
+                _ => {
+                    println!("Error setting context: {}", err);
+                    // Only assert for cases where we expect specific error messages
+                    // other errors can happen depending on the SELinux configuration
+                }
+            }
```

Comment:
I'm not sure about this entire test function (`test_selinux_context_setting`) as it will never fail.